lib/bitcoin/MongoDbStore.ts

```diff
@@ -0,0 +1,83 @@
+import { Collection, Db, MongoClient } from 'mongodb';
+
+/**
+ * Base class that contains the common MongoDB collection setup.
+ */
+export default class MongoDbStore {
+  /** Default database name used if not specified in constructor. */
+  public static readonly defaultDatabaseName: string = 'sidetree';
+  /** Collection name of this store. */
+  public readonly collectionName: string;
+  /** Database name used by this store. */
+  public readonly databaseName: string;
+
+  /** MondoDB instance. */
+  protected db: Db | undefined;
+  /** MongoDB collection */
+  protected collection: Collection<any> | undefined;
+
+  /**
+   * Constructs a `MongoDbStore`;
+   */
+  constructor (private serverUrl: string, collectionName: string, databaseName?: string) {
```

Comment:
Not sure if the `databaseName?` should be optional for this class?

Reply:
This follows the same pattern of all other `Store` classes, are you suggesting to change the pattern just for this class or for all?

Reply:
That's +2, so I will just change this class and leave other `Store` class alone. I see that those classes will inherit from this class in the future.